conjure-lib/src/main/java/com/palantir/conjure/java/lib/SafeLong.java

```diff
@@ -40,7 +40,7 @@ public long longValue() {
         return longValue;
     }
 
-    private static long check(long value) {
+    public static long check(long value) {
```

Comment:
Copied from https://github.com/palantir/conjure-java/pull/2389/files#r1811730849:
> I increased the visibility on this, because the logic is simple enough that to satisfy it is tantamount to copying the implementation detail (which felt bad). But also the logic is so simple that we aren't overly exposing some internals. Besides having more places rely on this is likely to prevent skew between implementations (IMO).

Reply:
I'm not sure I agree that we should expose apis for `SafeLong` using the `long` primitive type

Reply:
How would you suggest a non-boxing version of the check that happens during creation that also doesn't simply duplicate this logic?

Reply:
I don't think there's anything wrong with duplicating logic -- SafeLong provides two constants similarly to `Integer.MAX_VALUE` and `Integer.MIN_VALUE`.

Reply:
You against making the primitive constants public? Cause right now I have to compare it to a `SafeLong`. Not impossible just more awkward.

Reply:
Hmm, I'd prefer not to make those public because we'd lose some symmetry with the standard boxed primitive types from java.

Reply:
At a high level, I don't want to make changes to SafeLong lightly. My preference is not to expand in the process of implementing this feature -- if we find that there are many cases where enhancements are helpful, we should by all means make the improvements separately, then apply them to places like this which may benefit.